backend/routes/auth.js

```diff
@@ -543,22 +547,45 @@ router.put("/:account", middlewares.isAccountAdmin, updateUser);
 router.put("/:account/password", resetPassword);
 
 function createSession(place, req, res, next, user) {
-
 	req.session.regenerate(function(err) {
 		req[C.REQ_REPO].logger.logInfo("Creating session for " + " " + user.username);
 		if(err) {
 			responseCodes.respond(place, responseCodes.EXTERNAL_ERROR(err), res, {username: user.username});
 		} else {
 			req[C.REQ_REPO].logger.logDebug("Authenticated user and signed token.");
 
-			req.session[C.REPO_SESSION_USER] = user;
-			req.session.cookie.domain				 = config.cookie_domain;
+			user.socketId = req.headers[C.HEADER_SOCKET_ID];
+			user.webSession = false;
+
+			if (req.headers && req.headers["user-agent"]) {
+				user.webSession = !!req.headers["user-agent"];
+			}
 
+			req.session[C.REPO_SESSION_USER] = user;
+			req.session.cookie.domain = config.cookie_domain;
 			if (config.cookie.maxAge) {
 				req.session.cookie.maxAge = config.cookie.maxAge;
 			}
 
-			responseCodes.respond(place, req, res, next, responseCodes.OK, {username: user.username, roles: user.roles, flags: user.flags});
+			if (user.webSession) {
+				const query = {
+					"_id": { $ne: req.session.id },
+					"session.user.username": user.username,
+					"session.user.webSession": true
+				};
+
+				db.getCollection("admin", "sessions").then(_dbCol => {
```

Comment:
@sanmont3drepo is this how you'd do clear the session? considering we manage it through express-session, it feels  abit weird that we never store it but we manually delete it like this. Is there a function we're suppose to be using?

Also I think this should live in sessionManager? at the point of creating a new session, it removes the old ones? feels a bit coupled for it to be living in the routes.